### PR TITLE
IEP-462: Erase Flash Functionality added 

### PR DESCRIPTION
**Pull request by alirana01**

### CODE REVIEW

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+			try
+			{
+				deviceInformationText.setText(Messages.EraseFlashDialog_LoadingMessage);
+				infoThread = new Thread(this);
+				infoThread.start();
+			}
+			catch (Exception exception)
+			{
+				UIPlugin.getDefault().getLog().error("Exception while getting chip information", exception);
+			}
```
Comment (sigmaaa) [inline bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java:222]: externalize this error message. We can use our Logger class.

**Review by sigmaaa — CHANGES_REQUESTED** [review]
LGTM. We can incorporate the suggested changes.

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+		super(parent);
+		try
+		{
+			ports = SerialPort.list();
+			espToolCommands = new EspToolCommands();
+		}
+		catch (IOException e)
+		{
+			UIPlugin.getDefault().getLog().error("Exception while getting Ports", e);
+		}
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: @sigmaaa I have used our logger for logging and removed the message here, externalizing the exception message didnt felt like the right thing as at the end we have to debug these logs and externalizing them may lead to more confusions in log file.

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+		return super.close();
+	}
+
+	@Override
+	protected void okPressed()
+	{
+		setReturnCode(OK);
+		getButton(IDialogConstants.OK_ID).setEnabled(false);
+		deviceInformationText.setText("Erasing Flash....");
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+		Thread eraseFlashThread = new Thread(new Runnable()
+		{
+			@Override
+			public void run()
+			{
+				try
+				{
+					Display.getDefault().syncExec(() -> comPortsCombo.setEnabled(false));
+					Display.getDefault().syncExec(() -> deviceInformationText.setText(""));
+					Process eraseFlashProcess = espToolCommands.eraseFlash(selectedPort);
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+					InputStream targetIn = eraseFlashProcess.getInputStream();
+					BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(targetIn));
+					String line = "";
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+					while ((line = bufferedReader.readLine()) != null)
+					{
+						final String toWrite = line;
+						Display.getDefault().asyncExec(() -> deviceInformationText.append(toWrite + "\n"));
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+					}
+				}
+				catch (Exception e)
+				{
+					UIPlugin.getDefault().getLog().error("Exception while getting erasing flash", e);
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+				int readCharInt = 0;
+				while ((readCharInt = bufferedReader.read()) != -1)
+				{
+					final char charToWrite = (char) readCharInt;
+					Display.getDefault().syncExec(() -> deviceInformationText.append(Character.toString(charToWrite)));
+				}
+			}
+			catch (Exception e)
+			{
+				UIPlugin.getDefault().getLog().error("Exception while getting chip information", e);
```
[the remark this answers is not in the document]
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java (hunk 1 above)
Comment (alirana01) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/dialogs/EraseFlashDialog.java]: Done

**Review by sigmaaa — APPROVED** [review]

**Pull request merged**